src/plugins/dataflow/components/dataflow-tile.tsx

```diff
@@ -164,13 +175,14 @@ export default class DataflowToolComponent extends BaseComponent<IProps, IDatafl
     // 0 - cleared, ready to record
     // 1 - recording in progress
     // 2 - stopped, ready to clear
```

Comment:
I did this in the conditionals at the top of the version of the `tick` function that I pushed, but I think we should make it clear everywhere where these states are what they are.

```
0 program: executing, dataSet: empty
1 program: executing, dataSet: writing in progress
2 program: not executing,  dataSet: populated

//below are "substates" of #2 above
isPlaying: playbackIndex incrementing, Nodes updated "by hand" rather than via execution
isPaused: playbackIndex not incrementing, nodes stay as they were at last index above
```
I wonder if all of this could/should be consolidated into a single piece of state - Maybe that would make the current organization more difficult, curious to know what you think. 

 (it will, either way, eventually, but not for this PR need to be on the model so that a person could leave their "isRecorded" program in there and return to it the next day if they wanted).

Reply:
Are you saying perhaps have 4 levels of the program (0, 1, 2, 3) where 1, and 2 account for Recording and Playback respectively? I suppose that could work and you could get rid of isPlaying, but I don't think its too confusing the way it's written atm.

Reply:
I replaced my comments with yours.

Reply:
I think that if `isPlaying` is the only one of the states that is toggle-able and is guaranteed to only occur when recording mode is `2` then it makes sense to keep it as it is right now.  Was just trying to make sure I understood the states correctly, and that the code doesn't allow for an invalid combination of the two states.